assisted-inject-processor/src/main/java/com/squareup/inject/assisted/processor/AssistedInjectProcessor.kt

```diff
@@ -244,7 +252,7 @@ class AssistedInjectProcessor : AbstractProcessor() {
       throw StopProcessingException("Factory interface defines multiple factory methods.", factory)
     }
     val method = methods.single()
-    if (!types.isAssignable(type.asType(), method.returnType)) {
+    if (!types.isAssignable(types.erasure(type.asType()), types.erasure(method.returnType))) {
```

Comment:
I'm confused what the intention is, but my guess is that `type.asType()` is just the type of the parameterized type element, not the resolved type? This a total guess just looking at the snippet here and nothing else. 

Also the order of the `Types` methods always confuses me. Maybe they're reversed?